app/(marketing)/layout.tsx

```diff
@@ -13,21 +13,25 @@ interface MarketingLayoutProps {
 export default async function MarketingLayout({
   children,
 }: MarketingLayoutProps) {
+  const user = await getCurrentUser()
+
   return (
     <div className="flex min-h-screen flex-col">
       <header className="container z-40 bg-background">
         <div className="flex h-20 items-center justify-between py-6">
           <MainNav items={marketingConfig.mainNav} />
           <nav>
-            <Link
-              href="/login"
-              className={cn(
-                buttonVariants({ variant: "secondary", size: "sm" }),
-                "px-4"
-              )}
-            >
-              Login
-            </Link>
+            {user && (
```

Comment:
You are doing the opposite thing. you are checking if the user is logged in, then displaying the login button.  Instead you should do like this suggestion
```suggestion
            {!user && (
```